apps/angular/styling/src/app/text.component.ts

```diff
@@ -1,16 +1,36 @@
 /* eslint-disable @angular-eslint/component-selector */
-import { Component, Input } from '@angular/core';
+import { Component } from '@angular/core';
 
 @Component({
   selector: 'text',
   standalone: true,
   template: `
-    <p style="font-size: {{ font }}px; color: {{ color }}">
+    <p>
       <ng-content></ng-content>
     </p>
   `,
+  styles: `
+    :host {
+      &:not(.blue-text) {
+        font-size: 10px;
+        color: black;
+      }
+
+      &.blue-text {
+        font-size: 15px;
+        color: blue;
+      }
+    }
+
+    :host-context(.error) p {
+      font-size: 30px;
+      color: red;
+    }
+
+    :host-context(.warning) p {
+      font-size: 25px;
+      color: orange;
+    }
```

Comment:
the goal is to use css variables to be as customizable as possible.

Reply:
done